src/scalevoid.h

```diff
@@ -36,17 +36,28 @@ void checkWeight() {
  * @brief Initialize scale
  */
 void initScale() {
-    LoadCell.begin();
-    long stabilizingtime = 2000; // tare preciscion can be improved by adding a few seconds of stabilizing time
+    String scaleCalibrationString = number2string(scaleCalibration);
+    String StringCompare = "1.00";
+    boolean calibration = false;
+    LoadCell.begin(128);
+    long stabilizingtime = 4000; // tare preciscion can be improved by adding a few seconds of stabilizing time
     boolean _tare = true; //set this to false if you don't want tare to be performed in the next step
 
     u8g2.clearBuffer();
     u8g2.drawStr(0, 2, "Taring scale,");
     u8g2.drawStr(0, 12, "remove any load!");
-    u8g2.drawStr(0, 22, "....");
-    delay(2000);
+    
+    if(scaleCalibrationString == StringCompare){
```

Comment:
why is this comparison done with strings? should be numbers if possible to save memory

Reply:
Problem was, that you cant compare floats. Better solution would be to transform to double.

Reply:
can't use double als the calibration Value needs to be float.
So saved value as float but cast to double in if statement

Reply:
You can compare floats of course. Doubles are unnecessary here because you don't need that high accuracy

Reply:
if you need to check for equality you usually do something like this:
if(fabs(float1 - float2) < 0.00001)

Reply:
Is this solution acceptable? 
if ((double) scaleCalibration == 1.00)

Reply:
no, it's not guaranteed that scaleCalibration is exactly the double number 1.00. It could be 0.999999. Only if there is code that sets it to 1.00 you can assume that. 
You always need to compare the difference with a small epsilon like above otherwise

Reply:
It is set to 1.00 via mqtt in the process.
I am using this code for a few days and it is working flawless.

When you set it to 1.00 via mqtt the calibration process is started and an this process it is set to the correct value.
In the hx711-lib it is also advised to use 1.00 as the inital value to start the calibration.

Shall I change it anyway? Then we would need another float variable which ist set to 1.00 for the if-statement.

Reply:
just use `if (fabs(scaleCalibration - 1) < 0.00001) {`

Reply:
Oh, thats nice! Thanks!

Reply:
or abs if you don't want to use a float variable - it's a bit weird that our code uses double for almost everything for no reason